bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -480,9 +517,25 @@ public String toString() {
      */
     public void asyncReadEntries(long firstEntry, long lastEntry,
                                  ReadCallback cb, Object ctx) {
+
         // Little sanity check
-        if (firstEntry < 0 || lastEntry > lastAddConfirmed
-                || firstEntry > lastEntry) {
+        if (firstEntry < 0 || firstEntry > lastEntry) {
+            cb.readComplete(BKException.Code.ReadException, this, null, ctx);
```

Comment:
I am +1 on the UncorrectParameterException. 
Use cases:
- using BK as data store the reader, my reader seeks direclty to a single entry and if the LAC is not already updated it is not possible to know what is the problem without checking the LAC but we are in asynchronous code and it is very tricky
- for newbies it is better to get a different error as the LAC protocol is not very sinple at first use

My current projects won't break. Anyway we can document better this change in release notes